microsoftgraph/msgraph-cli-archived
# Feat/national clouds

## Overview

**New Commands**
- `mg cloud select` for selecting a cloud
- `mg cloud add` for adding a cloud
-  `mg cloud show-current` for viewing the current cloud
- `mg cloud delete` for deleting user defined clouds

**Modified Commands**
- `mg cloud login` Now accepts **clientd_id & tenant_id**

### Demo

**Selecting a cloud**
![image](https://user-images.githubusercontent.com/12011447/100469143-c9bab580-30e6-11eb-8dd8-a82c9b83d3f9.png)


**Sample profile.json**
```json
{
  "cloud": {
    "name": "Kenya",
    "graph_endpoint": "https://graph.microsoft.com/v1.0",
    "azure_ad_endpoint": "https://graph.microsoftonline.com"
  },
  "user_defined_clouds": [
    {
      "Kenya": {
        "name": "Kenya",
        "graph_endpoint": "https://graph.microsoft.com/v1.0",
        "azure_ad_endpoint": "https://graph.microsoftonline.com"
      }
    }
  ]
}
```
### Notes
Extends the login command with an option to pass a client_id, `mg login --scopes user.read --client-id <client_id>`

## Testing Instructions

* Pull this PR
* Build & start the docker container, instructions are in README.md
* Run `mg cloud add --name <name> --endpoint <endpoint> --authority<authority> to add a new cloud`
* Run `mg cloud select` to select a cloud
* Run `mg cloud show-current` to see the current cloud
* Run `mg cloud delete` to delete a user defined cloud
* Run `mg cloud update` to update a cloud
